packages/expo-clipboard/android/src/main/java/expo/modules/clipboard/ClipboardModule.kt

```diff
@@ -3,39 +3,77 @@ package expo.modules.clipboard
 import android.content.Context
 import android.content.ClipData
 import android.content.ClipboardManager
+import android.os.Bundle
+import android.util.Log
+import expo.modules.core.utilities.ifNull
 
-import expo.modules.core.ExportedModule
-import expo.modules.core.ModuleRegistry
-import expo.modules.core.Promise
-import expo.modules.core.interfaces.ExpoMethod
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
 
-class ClipboardModule(context: Context) : ExportedModule(context) {
-  private val NAME = "ExpoClipboard"
+private const val moduleName = "ExpoClipboard"
+private const val clipboardChangedEventName = "onClipboardChanged"
+private val TAG = ClipboardModule::class.java.simpleName
 
-  override fun getName(): String {
-    return NAME
-  }
+class ClipboardModule : Module() {
+  override fun definition() = ModuleDefinition {
+    name(moduleName)
 
-  override fun onCreate(moduleRegistry: ModuleRegistry) {
-    ClipboardEventEmitter(context, moduleRegistry)
-  }
+    function("getStringAsync") {
+      val clip = clipboardManager?.primaryClip?.takeIf { it.itemCount >= 1 }
+      clip?.getItemAt(0)?.text ?: ""
+    }
+
+    function("setString") { content: String ->
+      val clip = ClipData.newPlainText(null, content)
+      clipboardManager?.setPrimaryClip(clip)
+    }
```

Comment:
Let's migrate to `setStringAsync` and provide clear deprecation message.

Reply:
Great! I'll do that in a separate PR, because this also needs iOS and JS changes.